spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestUpdate.java

```diff
@@ -288,34 +265,34 @@ public void testExplain() {
     sql("EXPLAIN UPDATE %s SET dep = 'invalid' WHERE true", commitTarget());
 
     Table table = validationCatalog.loadTable(tableIdent);
-    Assert.assertEquals("Should have 1 snapshot", 1, Iterables.size(table.snapshots()));
+    assertThat(table.snapshots()).as("Should have 1 snapshot").hasSize(1);
 
     assertEquals(
         "Should have expected rows",
         ImmutableList.of(row(1, "hr"), row(2, "hardware"), row(null, "hr")),
         sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", selectTarget()));
   }
 
-  @Test
+  @TestTemplate
   public void testUpdateEmptyTable() {
-    Assume.assumeFalse("Custom branch does not exist for empty table", "test".equals(branch));
+    assumeThat("test".equals(branch)).as("Custom branch does not exist for empty table").isFalse();
```

Comment:
same as previously mentioned. We want to avoid using isFalse / isTrue as much as possible, because it won't print enough context when the assertion/assumption ever fails

Reply:
Sure, thank you!